plotly/grid_objs/grid_objs.py

```diff
@@ -148,7 +150,21 @@ def __init__(self, columns_or_json, fid=None):
         ```
         """
         # TODO: verify that columns are actually columns
-        if isinstance(columns_or_json, dict):
+        if pd and isinstance(columns_or_json, pd.DataFrame):
+            column_names = [name for name in columns_or_json.columns]
+            duplicate_name = utils.get_first_duplicate(column_names)
+            if duplicate_name:
+                err = exceptions.NON_UNIQUE_COLUMN_MESSAGE.format(duplicate_name)
+                raise exceptions.InputError(err)
+
+            # create columns from dataframe
+            all_columns = []
+            for name in column_names:
+                all_columns.append(Column(columns_or_json[name].tolist(), name))
+            self._columns = list(all_columns)
```

Comment:
Why `list` this? You've already instantiated a new list above `all_columns = []`. I don't see why an additional shallow copy is necessary?